Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_System.m

```diff
@@ -478,8 +470,9 @@
     if (isDebugBuild()) {
         return "debug";
     }
-    if (hasEmbeddedMobileProvision()) {
-        return "enterprise";
+    SentryMobileProvisionParser *parser = [[SentryMobileProvisionParser alloc] init];
+    if ([parser hasEmbeddedMobileProvisionProfile]) {
+        return [parser mobileProvisionProfileProvisionsAllDevices] ? "enterprise" : "adhoc";
```

Comment:
`m`: It would be great to add test for this, but I know we don't have tests for this and it could require refactoring.